### PR TITLE
[WIP] ✨ Add tilt support & rework manager credentials

### PR DESCRIPTION
**Pull request by ncdc**
**What this PR does / why we need it**:
Add tilt support based on the new tilt workflow in cluster-api.

Remove tilt bits from Dockerfile, as they are now contained in
cluster-api's Tiltfile.

Switch to a new kustomize secretGenerator approach for managing AWS
credentials. There is now a hack script to generate
config/manager/credentials.env containing just the credentials, and this
file is in .gitignore so it won't be accidentally checked in.

This goes along with https://github.com/kubernetes-sigs/cluster-api/pull/1894

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by ncdc** [comment]
PTAL at the kustomize changes for manager credentials

**Comment by rudoi** [comment]
lgtm as well, i'm so glad this stuff is coming together.

**Comment by ncdc** [comment]
@fabriziopandini PTAL

**Comment by ncdc** [comment]
This breaks the quick start. HOLD

File: tilt-provider.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "name": "aws",
+    "config": {
+        "image": "gcr.io/k8s-staging-cluster-api-aws/cluster-api-aws-controller",
+        "ignore": [
+            "config/manager/credentials.env"
```
Comment (detiber) [inline tilt-provider.json]: Should this be `config/manager/credentials` instead?

Comment (ncdc) [reply tilt-provider.json]: It's actually not needed anymore, because I'm using `only` in the Tiltfile. Will remove.

**Comment by ncdc** [comment]
Re-enabling WIP as I need to not break the quickstart and clusterctl v2

**Comment by ncdc** [comment]
Closing in favor of #1432

**Pull request closed without merging**